lib/matplotlib/tests/test_backends_interactive.py

```diff
@@ -626,29 +626,47 @@ def _impl_test_interactive_timers():
     # We only want singleshot if we specify that ourselves, otherwise we want
     # a repeating timer
     import os
+    import sys
     from unittest.mock import Mock
     import matplotlib.pyplot as plt
     # increase pause duration on CI to let things spin up
     # particularly relevant for gtk3cairo
     pause_time = 2 if os.getenv("CI") else 0.5
+    expected_100ms_calls = int(pause_time / 0.1)
     fig = plt.figure()
     plt.pause(pause_time)
     timer = fig.canvas.new_timer(0.1)
     mock = Mock()
     timer.add_callback(mock)
     timer.start()
     plt.pause(pause_time)
-    timer.stop()
-    assert mock.call_count > 1
+    # NOTE: The timer is as fast as possible, but this is different between backends
+    #       so we can't assert on the exact number but it should be faster than 100ms
+    assert mock.call_count > expected_100ms_calls, \
+        f"Expected more than {expected_100ms_calls} calls, got {mock.call_count}"
+
+    # Test updating the interval updates a running timer
+    timer.interval = 100
+    mock.call_count = 0
+    plt.pause(pause_time)
+    # GTK4 on macos runners produces about 3x as many calls as expected
+    # It works locally and on Linux though, so only skip when running on CI
```

Comment:
This is really annoying to wrestle with CI sometimes. I have no idea why the macos CI runners would have ~60 calls, but my local version and the linux runners get the correct ~20 calls.

If anyone has better ideas for how to test these timers let me know too. Each new test/"pause" we add here adds `2 seconds * nbackends / njobs` to the test runs and it looks like this test is currently one of the longest running in the suite at around ~10s.

Reply:
Did `plt.pause` not run for the expected `pause_time`, perhaps?

Reply:
That is a good thought! That led me to a few refactors here which I pushed up as separate commits.
* I removed `plt.pause()` and went directly to `start_event_loop()` now in case there was additional overhead in any of the pause draw calls.
* I parallelized the tests now so we only run two "pauses" and test multiple things during each pause.
* I noticed that GTK doesn't implement its own `start_event_loop()` and our current default one in `backend_bases.py` could run longer than expected if `flush_events()` added time to the loop. So I changed this in the final commit to be dependent on the run-time, not the number of iterations of sleep.